app/src/main/res/layout/hints_summary.xml

```diff
@@ -15,53 +15,49 @@
       type="org.oppia.android.app.hintsandsolution.HintsViewModel" />
   </data>
 
-  <LinearLayout
+  <androidx.constraintlayout.widget.ConstraintLayout
     android:id="@+id/hint_summary_container"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:layout_marginTop="32dp"
-    android:orientation="vertical">
+    android:layout_marginTop="32dp">
 
     <androidx.constraintlayout.widget.ConstraintLayout
+      android:id="@+id/reveal_hint_container"
       android:layout_width="match_parent"
       android:layout_height="wrap_content"
       android:layout_marginStart="28dp"
       android:layout_marginEnd="28dp"
       android:layout_marginBottom="20dp"
-      android:orientation="horizontal">
+      app:layout_constraintBottom_toTopOf="@+id/hint_list_container"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent">
 
       <TextView
         android:id="@+id/hint_title"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
+        style="@style/Heading2"
         android:layout_gravity="center"
         android:layout_marginTop="12dp"
         android:layout_marginBottom="12dp"
-        android:fontFamily="sans-serif-medium"
-        android:textColor="@color/oppiaPrimaryText"
-        android:textSize="20sp"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
       <Button
         android:id="@+id/reveal_hint_button"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:minWidth="160dp"
-        android:minHeight="48dp"
+        style="@style/StateButtonActive"
         android:layout_marginStart="8dp"
         android:layout_marginTop="16dp"
-        android:layout_marginBottom="32dp"
         android:layout_marginEnd="8dp"
+        android:layout_marginBottom="32dp"
         android:background="@drawable/reveal_hint_button_blue_background"
-        android:textSize="14sp"
+        android:minWidth="160dp"
+        android:minHeight="48dp"
         android:text="@string/reveal_hint"
-        android:textColor="@color/white"
+        android:visibility="@{viewModel.isHintRevealed? View.GONE : View.VISIBLE}"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/hint_title"
-        android:visibility="@{viewModel.isHintRevealed? View.GONE : View.VISIBLE}" />
+        app:layout_constraintTop_toBottomOf="@+id/hint_title" />
 
       <FrameLayout
```

Comment:
Out of curiosity, do we actually need this FrameLayout or can we move the constraint factors down to the ImageView it contains? This change + my comment above would result in a fully flattened layout which would be great performance-wise.

Reply:
The `FrameLayout` wasn't needed. Done as suggested.